keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -146,10 +146,16 @@ class CombinedRepository(
    * Removes artifacts, environments, and resources that were present in the [old]
    * delivery config and are not present in the [new] delivery config
    */
-  override fun removeResources(old: DeliveryConfig, new: DeliveryConfig) {
+  override fun removeDependents(old: DeliveryConfig, new: DeliveryConfig) {
     val newResources = new.resources.map { it.id }
+
     old.artifacts.forEach { artifact ->
-      if (artifact !in new.artifacts) {
+      val stillPresent = new.artifacts.any {
+        it.name == artifact.name &&
```

Comment:
Should we maybe want to extract it into an `equals` method or something? I suspect we might want to use it in the future as well.

Reply:
I had initially overridden the `equals()` method but feared it may have unintended side-effects, and ended up adding the filter only in this spot. I propose we revisit this if and when other cases come up.

Reply:
sounds good, thanks!